cmd/api/database/database.go

```diff
@@ -1,76 +1,90 @@
+	sqlDB, err := sql.Open("postgres", connStr)
 	if err != nil {
-		return db, fmt.Errorf("connecting to db: %w", err)
+		return nil, fmt.Errorf("connecting to db: %w", err)
 	}
 
-	err = db.Ping()
+	err = sqlDB.Ping()
 	if err != nil {
-		return db, fmt.Errorf("connecting to db: %w", err)
+		return nil, fmt.Errorf("connecting to db: %w", err)
 	}
 
-	fmt.Println("Successfully connected!")
-	return db, nil
+	log.Println("Successfully connected!")
+	return sqlDB, nil
 }
 
-var mGlobal *migrate.Migrate
+func NewStore(db *sql.DB) *Store {
+	CurrentStore := &Store{db: db}
+	return CurrentStore
+}
 
-func migrationUp() error {
-	driver, err := postgres.WithInstance(dbObjectGlobal, &postgres.Config{})
+func MigrationUp(store *Store, path string) error {
+	driver, err := postgres.WithInstance(store.db, &postgres.Config{})
 	if err != nil {
 		return fmt.Errorf("migrating up: %w", err)
 	}
 
-	path := os.Getenv("DATABASE_MIGRATIONS_PATH")
-	mGlobal, err = migrate.NewWithDatabaseInstance(
+	m, err := migrate.NewWithDatabaseInstance(
 		fmt.Sprintf("file://%s", path),
 		"postgres", driver)
 	if err != nil {
 		return fmt.Errorf("migrating up: %w", err)
 	}
 
-	err = mGlobal.Up()
+	err = m.Up()
 	if err != nil {
 		return fmt.Errorf("migrating up: %w", err)
 	}
 	return nil
 }
 
-/* Searches a book in database based on ID and returns it if succeed. */
-func searchById(id uuid.UUID) (Book, error) {
-	sqlStatement := `SELECT id, name, price, inventory, created_at, updated_at, archived FROM bookstable WHERE id=$1;`
-	foundRow := dbObjectGlobal.QueryRow(sqlStatement, id)
-	var bookToReturn Book
-	err := foundRow.Scan(&bookToReturn.ID, &bookToReturn.Name, &bookToReturn.Price, &bookToReturn.Inventory, &bookToReturn.CreatedAt, &bookToReturn.UpdatedAt, &bookToReturn.Archived)
+/* Change the status of 'archived' column on database. */
+func (store *Store) SetBookArchiveStatus(id uuid.UUID, archived bool) (book.Book, error) {
+	sqlStatement := `
+	UPDATE bookstable 
+	SET archived = $2
+	WHERE id = $1
+	RETURNING *`
+	updatedRow := store.db.QueryRow(sqlStatement, id, archived)
+	var bookToReturn book.Book
+	err := updatedRow.Scan(&bookToReturn.ID, &bookToReturn.Name, &bookToReturn.Price, &bookToReturn.Inventory, &bookToReturn.CreatedAt, &bookToReturn.UpdatedAt, &bookToReturn.Archived)
 	if err != nil {
 		switch err {
 		case sql.ErrNoRows:
-			return Book{}, fmt.Errorf("searching by ID: %w", errResponseBookNotFound)
+			return book.Book{}, fmt.Errorf("archiving on db: %w", book.ErrResponseBookNotFound)
 		default:
-			return Book{}, fmt.Errorf("searching by ID: %w", err)
+			return book.Book{}, fmt.Errorf("archiving on db: %w", err)
 		}
 	}
 
 	return bookToReturn, nil
 }
 
-func countRows(name string, minPrice32, maxPrice32 float32, archived bool) (int, error) {
+/* Counts how many rows in db fit the specified filter parameters. */
+func (store *Store) CountRows(name string, minPrice32, maxPrice32 float32, archived bool) (int, error) {
```

Comment:
This method need to be closer to the `ListBooks` method.
I think a better name would be `CountListBooksTotals` or simply `ListBooksTotals`.

Reply:
Done.